kong/global.lua

```diff
@@ -167,16 +167,42 @@ end
 function _GLOBAL.init_worker_events()
   -- Note: worker_events will not work correctly if required at the top of the file.
   --       It must be required right here, inside the init function
-  local worker_events = require "resty.worker.events"
+  local worker_events
+  local opts
 
-  local ok, err = worker_events.configure {
-    shm = "kong_process_events", -- defined by "lua_shared_dict"
-    timeout = 5,            -- life time of event data in shm
-    interval = 1,           -- poll interval (seconds)
+  local configuration = kong.configuration
 
-    wait_interval = 0.010,  -- wait before retry fetching event data
-    wait_max = 0.5,         -- max wait time before discarding event
-  }
+  if configuration and configuration.legacy_worker_events then
+
+    opts = {
+      shm = "kong_process_events", -- defined by "lua_shared_dict"
+      timeout = 5,            -- life time of event data in shm
+      interval = 1,           -- poll interval (seconds)
+
+      wait_interval = 0.010,  -- wait before retry fetching event data
+      wait_max = 0.5,         -- max wait time before discarding event
+    }
+
+    worker_events = require "resty.worker.events"
```

Comment:
I'd rather have the two `require`s at the top of the file, even if that means having to use some temp var names here.

Reply:
worker_events will not work correctly if required at the top of the file, It must be required right here, inside the init function